hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -777,6 +777,16 @@ final class VCFLine(
     }
   }
 
+  def infoToFloat(s: String): Float = {
+    s match {
+      case "nan" => Float.NaN
+      case "-nan" => Float.NaN
+      case "inf" => Float.PositiveInfinity
+      case "-inf" => Float.NegativeInfinity
+      case _ => s.toFloat
+    }
+  }
+
```

Comment:
I don't think this is the best solution. The fast path here is that `toFloat` succeeds. And furthermore, this doesn't handle the full inf/nan regex `[-+]?(INF|INFINITY|NAN)`.

Reply:
Of course, what do I see but 4 year ago me, implementing exactly this to keep up with observed behavior of the old line parser.

Reply:
We remain dependent on htsjdk for several features like tabix, we should probably just use their double parser, it's public. We can use it for both floats and doubles since we should be able to convert to a float after parsing a double.

https://github.com/samtools/htsjdk/blob/6d3fc7bc1f613ecfce1c22d368f3ae17cb86823d/src/main/java/htsjdk/variant/vcf/VCFUtils.java#L260-L280